frontend/src/scenes/data-warehouse/new/NewSourceWizard.tsx

```diff
@@ -232,9 +232,25 @@ function FirstStep(): JSX.Element {
 }
 
 function SecondStep(): JSX.Element {
-    const { selectedConnector } = useValues(sourceWizardLogic)
+    const { selectedConnector, source } = useValues(sourceWizardLogic)
 
-    return selectedConnector ? <SourceForm sourceConfig={selectedConnector} /> : <DatawarehouseTableForm />
+    let showPrefix = true
+    let showSourceFields = true
+    if (selectedConnector?.name === 'Salesforce') {
+        if (source.payload.code && source.payload.subdomain) {
+            showPrefix = true
+            showSourceFields = false
+        } else {
+            showPrefix = false
+            showSourceFields = true
+        }
+    }
```

Comment:
Not the biggest fan of having this custom logic here - could we add this as a function to the `SOURCE_DETAILS` const in `sourceWizardLogic`? Pass in the `source` object and return whether we hide/show fields

Reply:
yep good call made it a config